src/Sarif.Converters/FortifyFprConverter.cs

```diff
@@ -116,7 +114,8 @@ public override void Convert(Stream input, IResultLogWriter output, OptionallyEm
                     InstanceId = _automationId + "/"
                 },
                 Tool = tool,
-                Invocations = new[] { _invocation }
+                Invocations = new[] { _invocation },
+                Resources = new Resources {  Rules = _ruleDictionary}
```

Comment:
>Rules [](start = 45, length = 5)

Doesn't this mean you'll emit
```
"resources": {
  "rules": {}
}
```
if the dictionary is empty, rather than omitting the "rules" object? (In fact, rather than omitting `"resources"` entirely?) How about:
```
var run = new Run
{
    ...
};

if (_ruleDictionary.Any())
{
    run.Resources = new Resources { Rules = _ruleDictionary };
}
```

Reply:
our deserialization logic can elide the resources object in cases where its contained strings and rules are empty, is my thought. this keeps the initialization logic in one place. eventually, we should consider having the OM initialize core objects automatically, to make object creation more concise. will require building out the ShouldSerializeXXX helpers exhaustively. All this is one reason I built that helper that automatically hydrates the OM with all possible items. This will help comprehensively test the helpers described above.

---
In reply to: [244521205](https://github.com/Microsoft/sarif-sdk/pull/1191#discussion_r244521205) [](ancestors = 244521205)